rct229/reports/ashrae901_2019_summary_report.py

```diff
@@ -1,18 +1,190 @@
+        not_applicable_line = f"|Not Applicable|{self.ruleset_outcome['All'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['Envelope'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['Lighting'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['Receptacles'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['Transformers'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['HVAC-WaterSide'][RCTOutcomeLabel.NOT_APPLICABLE]}|{self.ruleset_outcome['HVAC-Chiller'][RCTOutcomeLabel.NOT_APPLICABLE]}|"
+        undetermined_line = f"|Undetermined (manual review)|{self.ruleset_outcome['All'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['Envelope'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['Lighting'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['Receptacles'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['Transformers'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['HVAC-WaterSide'][RCTOutcomeLabel.UNDETERMINED]}|{self.ruleset_outcome['HVAC-Chiller'][RCTOutcomeLabel.UNDETERMINED]}|"
+
+        self.summary_report = re.sub("Replace-Rules", rules_line, self.summary_report)
+        self.summary_report = re.sub("Replace-Pass", pass_line, self.summary_report)
+        self.summary_report = re.sub("Replace-Fail", fail_line, self.summary_report)
+        self.summary_report = re.sub(
+            "Replace-Not_Applicable", not_applicable_line, self.summary_report
+        )
+        self.summary_report = re.sub(
+            "Replace-Undetermined", undetermined_line, self.summary_report
+        )
+
+        with open(report_dir, "w", encoding="utf-8") as output_report:
+            output_report.write(self.summary_report)
+
+    def _section_name_helper(self, rule_outcome):
+        section_no = rule_outcome.split("-")[0]
+        if f"Section: {self.section_dict[section_no]}" not in self.summary_report:
+            return f"""
+### Section: {self.section_dict[section_no]}
+            """
+        else:
+            return None
+
+    def ruleset_outcome_count_helper(self, rule_outcome_id, rule_outcome_result):
+        section_no = rule_outcome_id.split("-")[0]
+
+        def _ruleset_outcome_count_helper(section_number, rule_outcome_result):
+            for result in [
+                RCTOutcomeLabel.PASS,
+                RCTOutcomeLabel.FAILED,
+                RCTOutcomeLabel.UNDETERMINED,
+                RCTOutcomeLabel.NOT_APPLICABLE,
+            ]:
+                self.ruleset_outcome[section_number][result] += rule_outcome_result[
+                    result
+                ]
+                self.ruleset_outcome["All"][result] += rule_outcome_result[result]
+
+        if section_no == "5":
+            _ruleset_outcome_count_helper("Envelope", rule_outcome_result)
```

Comment:
I think in the rule outcome, there is a new key called 'ruleset_section_title'

We can possibly use ruleset_section_title as dictionary to replace the manual string labels here.
In that case, we can remove these if-else statements

Reply:
Great point! Updated!